qcodes/dataset/data_set_protocol.py

```diff
@@ -377,6 +377,9 @@ def export(
             ValueError: If the export data type is not specified or unknown,
                 raise an error
         """
+        if isinstance(path, str):
```

Comment:
Personally, I find path = Path(path) more readable, but I am sure you have a good motivation for the explicit isinstance.

Reply:
The reason is that it could be noneIt could be None